src/sources/opentelemetry/integration_tests.rs

```diff
@@ -97,6 +104,81 @@ async fn receive_logs_legacy_namespace() {
     .await;
 }
 
+#[tokio::test]
+async fn receive_trace() {
+    // generate a trace request
+    let req = ExportTraceServiceRequest {
+        resource_spans: vec![ResourceSpans {
+            resource: None,
+            scope_spans: vec![ScopeSpans {
+                scope: None,
+                spans: vec![Span {
+                    trace_id: (1..17).collect_vec(), //trace_id [u8;16]
+                    span_id: (1..9).collect_vec(), // span_id [u8;8]
+                    parent_span_id: (1..9).collect_vec(), // parent_span_id [u8;8]
+                    name: "span".to_string(),
+                    kind: 1,
+                    start_time_unix_nano: 1713525203000000000,
+                    end_time_unix_nano: 1713525205000000000,
+                    attributes: vec![],
+                    dropped_attributes_count: 0,
+                    events: vec![],
+                    dropped_events_count: 0,
+                    links: vec![],
+                    dropped_links_count: 0,
+                    status: None,
+                    trace_state: "".to_string(),
+                }],
+                schema_url: "".to_string(),
+            }],
+            schema_url: "".to_string(),
+        }],
+    };
+    let body = req.encode_to_vec();
+
+    assert_source_compliance(&SOURCE_TAGS, async {
+        wait_ready(otel_health_url()).await;
+
+        let config = OpentelemetryConfig {
+            grpc: GrpcConfig {
+                address: source_grpc_address().parse().unwrap(),
+                tls: Default::default(),
+            },
+            http: HttpConfig {
+                address: source_http_address().parse().unwrap(),
+                tls: Default::default(),
+                keepalive: Default::default(),
+            },
+            acknowledgements: Default::default(),
+            log_namespace: Default::default(),
+        };
+
+        let (sender, trace_output, _) = new_source(EventStatus::Delivered);
```

Comment:
This function will capture the `LOGS` stream from the source. This integration test is interested in the `TRACES` output. You will either want to pass the output we want to capture as a parameter to this function, or create a new `new_traces_source` function.